swim/update_rollup_test.go

```diff
@@ -99,11 +100,23 @@ func (s *RollupTestSuite) TestFlushesAfterInterval() {
 	})
 
 	s.NotNil(s.r.FlushTimer(), "expected timer to be set")
+
+	// This is a race condition which never got triggered yet. If the scheduler
```

Comment:
+1 for finding a potential race
+1 for Dans remarks; put a TODO and let's make another post-it for addressing the race. I think this test is addressing 2 different concerns and that it needs to be split up into 2 tests: 1) Flushing empties the buffer, 2) Enabling the background flusher flushes buffer after flushInterval. To prove #2 one can either separate the concerns of what it means to flush the buffer and what it means to trigger flushing after an interval OR, and more simply, just raise an event (invoke a delegate) after the flushing occurs and assert the event was emitted (or delegate called) but not asserting that the buffer is actually empty.